lib/index.js

```diff
@@ -205,8 +205,12 @@ UI.prototype.stopProgress = function() {
 UI.prototype.prompt = function(questions, callback) {
   var inquirer = require('inquirer');
 
-  // If no callback was provided, automatically return a promise
-  return inquirer.prompt(questions, callback);
+  return inquirer.prompt(questions).then(function (answer) {
+    if (callback) {
```

Comment:
for compatibility, we'll still check if a callback was provided, and invoke it in the `then` block.